src/webgpu/api/operation/command_buffer/image_copy.spec.ts

```diff
@@ -1472,7 +1519,7 @@ function* generateTestTextureSizes({
   const textureSize: [number, number, number] = [
     widthAtThisLevel << mipLevel,
     heightAtThisLevel << mipLevel,
-    _mipSizeInBlocks.depthOrArrayLayers,
+    _mipSizeInBlocks.depthOrArrayLayers << (dimension === '3d' ? mipLevel : 0),
   ];
```

Comment:
this could use helper `virtualMipSize`. textureSize[1] would be wrong for 1d if this generator shouldn't be used for 1d, add an assert about that

Reply:
What this function is doing is actually the inverse of `virtualMipSize`, given the physical size of a level (in blocks), it computes the size of the texture at level 0.